src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -0,0 +1,494 @@
+    result = []
+    if no_children:
+        if parent:
+            result = [registry for registry in connected_registry_list \
+                if registry.parent.id is not None and registry.parent.id.endswith(parent)]
+        else:
+            result = [registry for registry in connected_registry_list if not registry.parent.id]
+    elif parent:
+        family_tree = {}
+        for registry in connected_registry_list:
+            family_tree[registry.id] = {
+                "registry": registry,
+                "childs": []
+            }
+            if registry.name == parent:
+                root_parent_id = registry.id
+        for registry in connected_registry_list:
+            parent_id = registry.parent.id
+            if parent_id and not parent_id.isspace():
+                family_tree[parent_id]["childs"].append(registry.id)
+        result = _get_descendancy(family_tree, root_parent_id)
+    else:
+        result = connected_registry_list
+    return result
+
+
+def acr_connected_registry_show(cmd,
+                                client,
+                                connected_registry_name,
+                                registry_name,
+                                resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+    return client.get(resource_group_name, registry_name, connected_registry_name)
+
+
+def acr_connected_registry_install_info(cmd,
+                                        client,
+                                        connected_registry_name,
+                                        registry_name,
+                                        fresh_install=False,
+                                        resource_group_name=None):
+    registry, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+    connected_registry = acr_connected_registry_show(
+        cmd, client, connected_registry_name, registry_name, resource_group_name)
+    connected_registry_localhost = connected_registry.login_server.host
+    parent_gateway_endpoint = connected_registry.parent.sync_properties.gateway_endpoint
+    parent_id = connected_registry.parent.id
+    sync_token_name = connected_registry.parent.sync_properties.token_id.split('/tokens/')[1]
+    if parent_id:
+        parent = parent_id.split('/connectedRegistry/')[1]
+        parent = acr_connected_registry_show(
+            cmd, client, parent, registry_name, resource_group_name)
+        parent_registry_endpoint = parent.login_server.host
+    else:
+        parent_registry_endpoint = registry.login_server
+
+    if fresh_install:
+        from ._client_factory import cf_acr_token_credentials
+        from .token import acr_token_credential_generate
+        cred_client = cf_acr_token_credentials(cmd.cli_ctx)
+        poller = acr_token_credential_generate(
```

Comment:
>acr_token_credential_generate [](start = 17, length = 29)

regenerate the password will invalidate existing password and could impact the client using it, do you want to at least print out a warning message if previous password exists?

Reply:
I'm not sure if there's a way to check. But every time the password is created we print a warning saying the client should store the password. This follows the same behavior found in the token command